ssl/statem/extensions_clnt.c

```diff
@@ -1408,6 +1411,8 @@ int tls_parse_stoc_etm(SSL *s, PACKET *pkt, unsigned int context, X509 *x,
 int tls_parse_stoc_ems(SSL *s, PACKET *pkt, unsigned int context, X509 *x,
                        size_t chainidx, int *al)
 {
+    if (s->options & SSL_OP_NO_EXTENDED_MASTER_SECRET)
+        return 1;
```

Comment:
I'm not sure if this change is necessary. If the option is set then we will not have proposed EMS. If the server sends an unsolicited ems response then the extensions system should barf. So we should only ever get here if we proposed it - which means the option is not set.

Reply:
Correct. This is just safety, but not strictly necessary.

Reply:
I will point out that tls_parse_stoc_etm() - immediately above this function - does check for SSL_OP_NO_ENCRYPT_THEN_MAC. This behavior should fall under the same umbrella.